crates/autopilot/src/run_loop.rs

```diff
@@ -89,7 +90,43 @@ impl RunLoop {
 
     async fn single_run_(&self, id: AuctionId, auction: &Auction) {
         tracing::info!("solving");
-        let mut solutions = self.solve(auction, id).await;
+        let solutions = self.solve(auction, id).await;
+
+        // Validate solutions and filter out invalid ones.
+        let mut solutions = solutions
+            .into_iter()
+            .filter_map(|(index, response)| {
+                if response.solutions.is_empty() {
+                    tracing::debug!(driver = ?self.drivers[index].url, "driver sent zero solutions");
+                    return None;
+                }
+
+                Some(
+                    response
+                        .solutions
+                        .into_iter()
+                        .filter_map(|solution| {
+                            if solution.score == U256::zero() {
+                                tracing::warn!(
+                                    id = ?solution.id,
+                                    driver = ?self.drivers[index].url,
+                                    "driver sent solution with zero score",
+                                );
+                                None
+                            } else {
+                                Some((index, solution))
+                            }
+                        })
+                        .collect_vec(),
+                )
+            })
+            .flatten()
+            .collect_vec();
+
+        if solutions.is_empty() {
+            tracing::warn!(?id, "no solutions for auction");
```

Comment:
nit: this shouldn't be warn but info at most

Reply:
Updated.